nrf52840 fails to build when QSPI pins are not defined

### Operating System

Linux

### IDE version

Arduino CLI 0.30.0

### Board

Custom nrf52840

### BSP version

1.3.0

### SPIFlash Library version

3.11.0

### Sketch as attached file if not stock example

examples/flash_info will trigger this issue

### What happened ?

Compilation fails with the follow error messages if QSPI pins are not defined:

```
/Arduino/libraries/Adafruit_SPIFlash/src/qspi/Adafruit_FlashTransport_QSPI_NRF.cpp:32:36: error: 'PIN_QSPI_SCK' was not declared in this scope; did you mean 'PIN_SPI_SCK'?
   32 |     : Adafruit_FlashTransport_QSPI(PIN_QSPI_SCK, PIN_QSPI_CS, PIN_QSPI_IO0,
      |                                    ^~~~~~~~~~~~
      |                                    PIN_SPI_SCK
```

I expect that code should only be referenced if EXTERNAL_FLASH_USE_QSPI is defined.

I would suggest adding that define to Adafruit_FlashTransport_QSPI_NRF.cpp which resolves the issue:
```
#if defined(NRF52840_XXAA) && defined(EXTERNAL_FLASH_USE_QSPI)
```

### How to reproduce ?

Create a custom board where PIN_QSPI_SCK is not defined in variant.h, compilation will fail with the described message.

### Debug Log as attached txt file

_No response_

### Screenshots

_No response_

--- a/src/qspi/Adafruit_FlashTransport_QSPI_NRF.cpp
+++ b/src/qspi/Adafruit_FlashTransport_QSPI_NRF.cpp
@@ -28,6 +28,32 @@
 #include "nrfx_qspi.h"
 #include <Arduino.h>
 
+// default PIN_QSPI_SCK, CS, IO0, IO1, IO2, IO3 to -1 to build with non-qpsi
+// board
+#ifndef PIN_QSPI_SCK
+#define PIN_QSPI_SCK -1
+#endif
+
+#ifndef PIN_QSPI_CS
+#define PIN_QSPI_CS -1
+#endif
+
+#ifndef PIN_QSPI_IO0
+#define PIN_QSPI_IO0 -1
+#endif
+
+#ifndef PIN_QSPI_IO1
+#define PIN_QSPI_IO1 -1
+#endif
+
+#ifndef PIN_QSPI_IO2
+#define PIN_QSPI_IO2 -1
+#endif
+
+#ifndef PIN_QSPI_IO3
+#define PIN_QSPI_IO3 -1
+#endif
+
 Adafruit_FlashTransport_QSPI::Adafruit_FlashTransport_QSPI(void)
     : Adafruit_FlashTransport_QSPI(PIN_QSPI_SCK, PIN_QSPI_CS, PIN_QSPI_IO0,
                                    PIN_QSPI_IO1, PIN_QSPI_IO2, PIN_QSPI_IO3) {}
@@ -91,8 +117,9 @@ void Adafruit_FlashTransport_QSPI::setClockSpeed(uint32_t clock_hz,
   // delay is set to one freq period
   uint8_t delay = 1;
 
-  if (clkdiv)
+  if (clkdiv) {
     delay = (1UL << (clkdiv - 1));
+  }
 
   NRF_QSPI->IFCONFIG1 &=
       ~(QSPI_IFCONFIG1_SCKFREQ_Msk | QSPI_IFCONFIG1_SCKDELAY_Msk);

